src/clp_ffi_js/ir/StreamReader.cpp

```diff
@@ -150,43 +128,19 @@ auto StreamReader::decode_range(size_t begin_idx, size_t end_idx) const -> Decod
             m_encoded_log_events.begin()
                     + static_cast<decltype(m_encoded_log_events)::difference_type>(end_idx)
     };
-    std::string message;
-    constexpr size_t cDefaultReservedMessageLength{512};
-    message.reserve(cDefaultReservedMessageLength);
     size_t log_num{begin_idx + 1};
     auto const results{emscripten::val::array()};
     for (auto const& log_event : log_events_span) {
-        message.clear();
-
-        auto const parsed{log_event.get_message().decode_and_unparse()};
-        if (false == parsed.has_value()) {
+        auto const json{log_event.serialize_to_json()};
```

Comment:
Here we retrieve the log event as a JSON since I am concerned about the cost of inserting nodes into an Embind object when we traverse the log event. Let me know if there're any concerns that I missed.

Reply:
Some benchmarking might be worthy before we decide on that though.